projects/igniteui-angular/src/lib/grids/state.directive.ts

```diff
@@ -206,13 +206,11 @@ export class IgxGridStateDirective {
             },
             restoreFeatureState: (context: IgxGridStateDirective, state: IColumnState[]): void => {
                 const newColumns = [];
-                const factory = context.resolver.resolveComponentFactory(IgxColumnComponent);
-                const groupFactory = context.resolver.resolveComponentFactory(IgxColumnGroupComponent);
                 state.forEach((colState) => {
                     const hasColumnGroup = colState.columnGroup;
                     delete colState.columnGroup;
                     if (hasColumnGroup) {
-                        const ref1 = groupFactory.create(context.viewRef.injector);
+                        const ref1 = context.viewRef.createComponent(IgxColumnGroupComponent, { injector: context.viewRef.injector});
```

Comment:
I believe the same thing like the pivot is true here as well - i.e. this will create elements next to the host of the state directive (the grid) in the DOM. Same solutions really - either a grid specified anchor or the standalone create method likely.